server/routes/messaging.ts

```diff
@@ -35,10 +37,7 @@ router.route("/api/messaging/coordinator-recruitment-text").get(
       }
     );
     if (!resp.ok) {
-      throw {
```

Comment:
I see you're removing the "throw" exceptions throughout and replacing with logger. I know I want logging to STDOUT so I can see them in the server log. But, is there a downside to not also throwing exceptions?

Reply:
This was a misunderstanding on my part. I should have clarified in my question about replacing it with the log. I'll add them throws back I removed.

Reply:
Yeah it would probably be better to keep some sort of "throw" because we want errors to propagate back to the frontend. Although, I actually don't think that it works correctly right now.

Reply:
@jasoncavanaugh Do we need to pass next in the async route callbacks to pass the errors to the errorHandler middleware?

Reply:
Reading the Express docs, this is how I understand using error handling middleware in asynchronous functions. Main differences: Wrapping the callback in a try-catch, passing the next function as a parameter and then calling it in the catch to pass an error when one is thrown.

![image](https://user-images.githubusercontent.com/74210169/224550418-207bf198-1391-41ef-a151-04ec92740b90.png)

Or is this what Express-Async-Handler is supposed to do?

Reply:
The asyncHandler is so that we can await promises in our endpoints. Express doesn't allow async/await out of the box, so everything needs to be wrapped with the asyncHandler.

As for how you've structured things here, I'm not sure that the try catch is necessary. From my understanding, if you throw any kind of error, it should propagate to the errorMiddleware file, which has a small amount of code for sending the error response back to the frontend. My original intention behind error handling was that we would expose different kinds of errors at different layers. Like if a request from the backend to Airtable failed with a 401 Unauthorized, we would want to log that information on the backend. But I was thinking we would propogate something like 500 Internal Server error to the frontend instead of the actual error from Airtable since it seemed "safer" to not expose that error to the client. But honestly, I just tried to get something set up, and I'm not too sure what best practices are when it comes to error handling anyways. I'd be curious to hear if Matt has any thoughts on the matter.

Reply:
@zuechai Is there any difference between `logger.error(new Error("message"))` and `logger.error("message")`? They both seemed to log the same thing when I compared them.

Reply:
You're right that they do the same thing. Looking at it now, I don't know why I thought it needed an instance of an Error class to include the trace. However, that's not true when I uncommented the other formatting.

I see how asyncHandler works now. The catch would be totally unnecessary as it removes the need for a try-catch and passing it to next(). The doc for that package sums it up concisely.